### ISSUE TITLE
When running composer self-update an error is shown referring to Icu4c and the self-update fails to run

### ISSUE THREAD

**Issue by rpkoller** [Homebrew/homebrew-core#25971]
**Please note we will close your issue without comment if you delete, do not read or do not fill out the issue checklist below and provide ALL the requested information. If you repeatedly fail to use the issue template, we will block you from ever submitting issues to Homebrew again.**

- [x] are reporting a bug others will be able to reproduce and not asking a question. If you're not sure or want to ask a question do so on our Discourse: https://discourse.brew.sh

looks like a bug

- [ ] have a problem with `brew install` (or `upgrade`, `reinstall`) a single, official formula (not cask)? If it's a general `brew` problem please file this issue at Homebrew/brew: https://github.com/Homebrew/brew/issues/new. If it's a `brew cask` problem please file this issue at https://github.com/caskroom/homebrew-cask/issues/new. If it's a tap (e.g. Homebrew/homebrew-php) problem please file this issue at the tap.

no issues with brew install

- [x] ran `brew update` and can still reproduce the problem?

after `brew update`  the issue persists

- [x] ran `brew doctor`, fixed all issues and can still reproduce the problem?

the only warning shown are two unlinked kegs `mariadb` and `python@2

- [x] ran `brew gist-logs <formula>` (where `<formula>` is the name of the formula that failed) and included the output link?

```
brew gist-logs icu4c
Error: No logs.
```

- [x] if `brew gist-logs` didn't work: ran `brew config` and `brew doctor` and included their output with your issue?

```
Please note that these warnings are just used to help the Homebrew maintainers
with debugging if you file an issue. If everything you use Homebrew for is
working fine: please don't worry or file an issue; just ignore this. Thanks!

Warning: You have unlinked kegs in your Cellar
Leaving kegs unlinked can lead to build-trouble and cause brews that depend on
those kegs to fail to run properly once built. Run `brew link` on these:
  mariadb
  python@2
```

To help us debug your issue please explain:
- What you were trying to do (and why)
I was trying to run `composer self-update` 

- What happened (include command output)
```
$> composer self-update
dyld: Library not loaded: /usr/local/opt/icu4c/lib/libicui18n.60.dylib
  Referenced from: /usr/local/bin/php
  Reason: image not found
Trace/BPT trap: 5
```
- What you expected to happen
I've expected that `composer self-update` would work as usual 

- Step-by-step reproduction instructions (by running `brew install` commands)
Using the most current kegs `php 7.2.4` and `icu4c 61.1`

# Formula additions or changes
To get formulae added or changed in Homebrew please file a [Pull Request](https://github.com/Homebrew/homebrew-core/blob/master/CONTRIBUTING.md)
We will close issues requesting formulae changes.

**Comment by SMillerDev** [comment]
Do you have composer installed through brew?

**Comment by rpkoller** [comment]
Sorry, I forgot to mention. No I haven't. I've installed it manually at `/usr/local/bin/composer` like recommended in the getting started guide on https://getcomposer.org

**Comment by ilovezfs** [comment]
You need to `brew upgrade`

**Comment by rpkoller** [comment]
@ilovezfs I did `brew update` and `brew upgrade` several times after the issues turned up, also right now after you recommended again, but still resulting in the initial error described :/

**Comment by SMillerDev** [comment]
what is your PHP version?

**Comment by rpkoller** [comment]
The version I've installed with homebrew: 

```
$> which php
/usr/local/bin/php
```

```
php -> ../Cellar/php/7.2.4/bin/php
```

**Comment by ilovezfs** [comment]
```
brew rm php
brew fetch -f php
brew install php
```

**Comment by rpkoller** [comment]
@ilovezfs thanks a lot! that worked!

```
$> php -v
PHP 7.2.4 (cli) (built: Mar 29 2018 15:25:47) ( NTS )
Copyright (c) 1997-2018 The PHP Group
Zend Engine v3.2.0, Copyright (c) 1998-2018 Zend Technologies
    with Zend OPcache v7.2.4, Copyright (c) 1999-2018, by Zend Technologies
```
```
$> composer self-update
You are already using composer version 1.6.3 (stable channel).
```
so it wasnt a bug in php nor Icu4c but just some sort of hickup in the update process for php? is there an easy way or lets better say indication to distinguish between a real bug and that sort of hickup? so i can save your time with that sort of support ticket in case something similar happens again? :)  thanks for the help!

**Comment by SMillerDev** [comment]
You can check if `php -v` reports the latest version and/or errors.

**Comment by ilovezfs** [comment]
>some sort of hickup in the update process for php

Yeah, the upgrade went through prematurely (before upstream php has designated it a final release), and was reverted. But before it was reverted, you installed it. At that time, it had been built against icu4c 60. Then later the php upgrade was designated a final release, and we built it again. In the interim, icu4c was upgraded to 61. So you still had the old one built against icu4c 60.
